### About Dataset

#### Context
This dataset contains records from the new crime incident report system, which includes a reduced set of fields focused on capturing the type of incident as well as when and where it occurred.

#### Content
This dataset has 260,760 rows and 17 columns:

- **INCIDENT_NUMBER**: 
- **OFFENSE_CODE**: 
- **OFFENSE_CODE_GROUP**: 
- **OFFENSE_DESCRIPTION**: 
- **DISTRICT**: 
- **REPORTING_AREA**: 
- **SHOOTING**: 
- **OCCURRED_ON_DATE**: 
- **YEAR**: 
- **MONTH**: 
- **DAY_OF_WEEK**: 
- **HOUR**: 
- **UCR_PART**: 
- **STREET**: 
- **LATITUDE**: 
- **LONGITUDE**: 
- **LOCATION**: 


In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


In [22]:
data = pd.read_csv(r"C:\Users\Anass\AppData\Roaming\Code\User\globalStorage\github.remotehub\ec024370594da218ecf3b0482736e621\changestore\vscode-vfs-github\anass1209\data-analytics-criminal-project\crime.csv", encoding = "iso_ir_100")

In [20]:
data1 = pd.read_csv(r"https://github.com/anass1209/data-analytics-criminal-project/raw/main/crime.csv", encoding = "iso_ir_100")


KeyboardInterrupt



In [23]:
data.head(10)

INCIDENT_NUMBER  OFFENSE_CODE               OFFENSE_CODE_GROUP  \
0      I182070945           619                          Larceny   
1      I182070943          1402                        Vandalism   
2      I182070941          3410                            Towed   
3      I182070940          3114             Investigate Property   
4      I182070938          3114             Investigate Property   
5      I182070936          3820  Motor Vehicle Accident Response   
6      I182070933           724                       Auto Theft   
7      I182070932          3301                  Verbal Disputes   
8      I182070931           301                          Robbery   
9      I182070929          3301                  Verbal Disputes   

                          OFFENSE_DESCRIPTION DISTRICT REPORTING_AREA  \
0                          LARCENY ALL OTHERS      D14            808   
1                                   VANDALISM      C11            347   
2                         TOWED MOTOR VEHICLE       D4            151   
3                        INVESTIGATE PROPERTY       D4            272   
4                        INVESTIGATE PROPERTY       B3            421   
5  M/V ACCIDENT INVOLVING PEDESTRIAN - INJURY      C11            398   
6                                  AUTO THEFT       B2            330   
7                              VERBAL DISPUTE       B2            584   
8                            ROBBERY - STREET       C6            177   
9                              VERBAL DISPUTE      C11            364   

  SHOOTING     OCCURRED_ON_DATE  YEAR  MONTH DAY_OF_WEEK  HOUR    UCR_PART  \
0      NaN  2018-09-02 13:00:00  2018      9      Sunday    13    Part One   
1      NaN  2018-08-21 00:00:00  2018      8     Tuesday     0    Part Two   
2      NaN  2018-09-03 19:27:00  2018      9      Monday    19  Part Three   
3      NaN  2018-09-03 21:16:00  2018      9      Monday    21  Part Three   
4      NaN  2018-09-03 21:05:00  2018      9      Monday    21  Part Three   
5      NaN  2018-09-03 21:09:00  2018      9      Monday    21  Part Three   
6      NaN  2018-09-03 21:25:00  2018      9      Monday    21    Part One   
7      NaN  2018-09-03 20:39:37  2018      9      Monday    20  Part Three   
8      NaN  2018-09-03 20:48:00  2018      9      Monday    20    Part One   
9      NaN  2018-09-03 20:38:00  2018      9      Monday    20  Part Three   

              STREET        Lat       Long                     Location  
0         LINCOLN ST  42.357791 -71.139371  (42.35779134, -71.13937053)  
1           HECLA ST  42.306821 -71.060300  (42.30682138, -71.06030035)  
2        CAZENOVE ST  42.346589 -71.072429  (42.34658879, -71.07242943)  
3         NEWCOMB ST  42.334182 -71.078664  (42.33418175, -71.07866441)  
4           DELHI ST  42.275365 -71.090361  (42.27536542, -71.09036101)  
5         TALBOT AVE  42.290196 -71.071590  (42.29019621, -71.07159012)  
6        NORMANDY ST  42.306072 -71.082733  (42.30607218, -71.08273260)  
7            LAWN ST  42.327016 -71.105551  (42.32701648, -71.10555088)  
8  MASSACHUSETTS AVE  42.331521 -71.070853  (42.33152148, -71.07085307)  
9          LESLIE ST  42.295147 -71.058608  (42.29514664, -71.05860832)

![Getting Started](image.png)

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 319073 entries, 0 to 319072
Data columns (total 17 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   INCIDENT_NUMBER      319073 non-null  object 
 1   OFFENSE_CODE         319073 non-null  int64  
 2   OFFENSE_CODE_GROUP   319073 non-null  object 
 3   OFFENSE_DESCRIPTION  319073 non-null  object 
 4   DISTRICT             317308 non-null  object 
 5   REPORTING_AREA       319073 non-null  object 
 6   SHOOTING             1019 non-null    object 
 7   OCCURRED_ON_DATE     319073 non-null  object 
 8   YEAR                 319073 non-null  int64  
 9   MONTH                319073 non-null  int64  
 10  DAY_OF_WEEK          319073 non-null  object 
 11  HOUR                 319073 non-null  int64  
 12  UCR_PART             318983 non-null  object 
 13  STREET               308202 non-null  object 
 14  Lat                  299074 non-null  float64
 15  Long             

In [30]:
data['REPORTING_AREA'].str.strip()

data = data[data['REPORTING_AREA'] != " "]

data['REPORTING_AREA'] = data['REPORTING_AREA'].astype('int')

In [31]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 298823 entries, 0 to 319072
Data columns (total 17 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   INCIDENT_NUMBER      298823 non-null  object 
 1   OFFENSE_CODE         298823 non-null  int64  
 2   OFFENSE_CODE_GROUP   298823 non-null  object 
 3   OFFENSE_DESCRIPTION  298823 non-null  object 
 4   DISTRICT             298793 non-null  object 
 5   REPORTING_AREA       298823 non-null  int32  
 6   SHOOTING             990 non-null     object 
 7   OCCURRED_ON_DATE     298823 non-null  object 
 8   YEAR                 298823 non-null  int64  
 9   MONTH                298823 non-null  int64  
 10  DAY_OF_WEEK          298823 non-null  object 
 11  HOUR                 298823 non-null  int64  
 12  UCR_PART             298733 non-null  object 
 13  STREET               297855 non-null  object 
 14  Lat                  296292 non-null  float64
 15  Long                 2

In [32]:
data['OCCURRED_ON_DATE'] = pd.to_datetime(data['OCCURRED_ON_DATE'])

In [33]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 298823 entries, 0 to 319072
Data columns (total 17 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   INCIDENT_NUMBER      298823 non-null  object        
 1   OFFENSE_CODE         298823 non-null  int64         
 2   OFFENSE_CODE_GROUP   298823 non-null  object        
 3   OFFENSE_DESCRIPTION  298823 non-null  object        
 4   DISTRICT             298793 non-null  object        
 5   REPORTING_AREA       298823 non-null  int32         
 6   SHOOTING             990 non-null     object        
 7   OCCURRED_ON_DATE     298823 non-null  datetime64[ns]
 8   YEAR                 298823 non-null  int64         
 9   MONTH                298823 non-null  int64         
 10  DAY_OF_WEEK          298823 non-null  object        
 11  HOUR                 298823 non-null  int64         
 12  UCR_PART             298733 non-null  object        
 13  STREET             

In [34]:
data.isnull().sum()

INCIDENT_NUMBER             0
OFFENSE_CODE                0
OFFENSE_CODE_GROUP          0
OFFENSE_DESCRIPTION         0
DISTRICT                   30
REPORTING_AREA              0
SHOOTING               297833
OCCURRED_ON_DATE            0
YEAR                        0
MONTH                       0
DAY_OF_WEEK                 0
HOUR                        0
UCR_PART                   90
STREET                    968
Lat                      2531
Long                     2531
Location                    0
dtype: int64

In [35]:
data.describe()

OFFENSE_CODE  REPORTING_AREA               OCCURRED_ON_DATE  \
count  298823.000000   298823.000000                         298823   
mean     2291.362040      383.211132  2017-01-21 10:11:06.319198720   
min       111.000000        0.000000            2015-06-15 00:00:00   
25%       802.000000      177.000000            2016-04-04 09:00:00   
50%      2907.000000      344.000000            2017-01-23 11:06:00   
75%      3201.000000      544.000000            2017-11-03 20:54:00   
max      3831.000000      962.000000            2018-09-03 21:25:00   
std      1183.114445      242.286937                            NaN   

                YEAR          MONTH           HOUR            Lat  \
count  298823.000000  298823.000000  298823.000000  296292.000000   
mean     2016.549292       6.611623      13.124923      42.315233   
min      2015.000000       1.000000       0.000000      -1.000000   
25%      2016.000000       4.000000       9.000000      42.297521   
50%      2017.000000       7.000000      14.000000      42.325538   
75%      2017.000000       9.000000      18.000000      42.348624   
max      2018.000000      12.000000      23.000000      42.395042   
std         1.001150       3.280769       6.280909       0.552286   

                Long  
count  296292.000000  
mean      -71.071522  
min       -71.178674  
25%       -71.097200  
50%       -71.077610  
75%       -71.062563  
max        -1.000000  
std         0.892443

### Interpretations of Statistics

- **Lat and Long**:
  - The values in the `Lat` and `Long` columns are concentrated around the mean, with relatively low standard deviations (0.55 for `Lat` and 0.89 for `Long`). This indicates that latitude and longitude coordinates are generally clustered around the average values.
  - The count of non-null values is lower in these columns compared to others, suggesting the presence of missing values.

- **YEAR**:
  - The mean is 2016.55 with a standard deviation of 1.00, showing that the years are concentrated around this average value with minimal variability.

- **MONTH**:
  - The mean is 6.61, indicating that most incidents occur around the middle of the year. The standard deviation of 3.28 shows moderate variability in the months.

- **HOUR**:
  - The mean is 13.12, which means incidents tend to occur in the afternoon, with a concentration around this time of day.

- **OFFENSE_CODE**:
  - The average offense code is 2,291.36, indicating that offense codes generally tend to be around this number.


In [36]:
data.describe(include= 'object')

INCIDENT_NUMBER               OFFENSE_CODE_GROUP OFFENSE_DESCRIPTION  \
count           298823                           298823              298823   
unique          264893                               67                 241   
top         I162030584  Motor Vehicle Accident Response  INVESTIGATE PERSON   
freq                13                            30558               18067   

       DISTRICT SHOOTING DAY_OF_WEEK    UCR_PART         STREET  \
count    298793      990      298823      298733         297855   
unique       12        1           7           4           3944   
top          B2        Y      Friday  Part Three  WASHINGTON ST   
freq      46561      990       45445      146858          14144   

                        Location  
count                     298823  
unique                     17731  
top     (0.00000000, 0.00000000)  
freq                        2531

### Interpretations of Statistics

- **Lat and Long**:
  - The values in the `Lat` and `Long` columns are concentrated around the mean, with relatively low standard deviations (0.55 for `Lat` and 0.89 for `Long`). This indicates that latitude and longitude coordinates are generally clustered around the average values.
  - The count of non-null values is lower in these columns compared to others, suggesting the presence of missing values.

- **YEAR**:
  - The mean is 2016.55 with a standard deviation of 1.00, showing that the years are concentrated around this average value with minimal variability.

- **MONTH**:
  - The mean is 6.61, indicating that most incidents occur around the middle of the year. The standard deviation of 3.28 shows moderate variability in the months.

- **HOUR**:
  - The mean is 13.12, which means incidents tend to occur in the afternoon, with a concentration around this time of day.

- **OFFENSE_CODE**:
  - The average offense code is 2,291.36, indicating that offense codes generally tend to be around this number.


" Which offense groups are the most frequently occurring crimes? "

In [39]:
data['OFFENSE_CODE_GROUP'].value_counts().head(5)

OFFENSE_CODE_GROUP
Motor Vehicle Accident Response    30558
Larceny                            25256
Medical Assistance                 22491
Investigate Person                 18063
Other                              17157
Name: count, dtype: int64

In [41]:
data['OFFENSE_CODE_GROUP'].value_counts().tail()

OFFENSE_CODE_GROUP
Manslaughter                                 6
INVESTIGATE PERSON                           4
Biological Threat                            2
HUMAN TRAFFICKING - INVOLUNTARY SERVITUDE    2
Burglary - No Property Taken                 2
Name: count, dtype: int64

In [47]:
data.nunique()

INCIDENT_NUMBER        264893
OFFENSE_CODE              221
OFFENSE_CODE_GROUP         67
OFFENSE_DESCRIPTION       241
DISTRICT                   12
REPORTING_AREA            878
SHOOTING                    1
OCCURRED_ON_DATE       219778
YEAR                        4
MONTH                      12
DAY_OF_WEEK                 7
HOUR                       24
UCR_PART                    4
STREET                   3944
Lat                     17717
Long                    17715
Location                17731
dtype: int64